debezium-core/src/main/java/io/debezium/transforms/DecodeLogicalDecodingMessageContent.java

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms;
+
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.util.Map;
+import java.util.Objects;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.connect.components.Versioned;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.errors.DataException;
+import org.apache.kafka.connect.transforms.ReplaceField;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import io.debezium.Module;
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.config.Configuration;
+import io.debezium.data.Envelope;
+import io.debezium.schema.FieldNameSelector;
+import io.debezium.transforms.outbox.EventRouterConfigDefinition;
+import io.debezium.transforms.outbox.JsonSchemaData;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+/**
+ * The transform converts binary content of a logical decoding message to a structured form.
+ * One of the possible usages is to apply the transform before {@link io.debezium.transforms.outbox.EventRouter} so that the transform
+ * will produce a record suitable for the Outbox SMT.
+ *
+ * @author Roman Kudryashov
+ */
+public class DecodeLogicalDecodingMessageContent<R extends ConnectRecord<R>> implements Transformation<R>, Versioned {
```

Comment:
Would it make more senses for this to be in `debezium-connector-postgres` rather than core?

Reply:
done